backend/devices/bitbox02/device.go

```diff
@@ -209,24 +215,28 @@ func (device *Device) SetDeviceName(deviceName string) error {
 }
 
 // DeviceInfo retrieves the current device info from the bitbox
-func (device *Device) DeviceInfo() (string, error) {
+func (device *Device) DeviceInfo() (*DeviceInfo, error) {
 	request := &messages.Request{
 		Request: &messages.Request_DeviceInfo{
 			DeviceInfo: &messages.DeviceInfoRequest{},
 		},
 	}
 
+	deviceInfo := &DeviceInfo{}
+
 	response, err := device.query(request)
 	if err != nil {
-		return "", err
+		return deviceInfo, err
```

Comment:
`return nil, err` (if err != nil, all other return values are to be ignored anyway).

then you can define the above `deviceInfo` further down.